File: Day05/Easy/Solution.java
import java.util.*;

class Solution {
    public boolean armstrongNumber(int n) {
        int sum = 0, x = n;
        while (x > 0) {
            int d = x % 10;
            sum += Math.pow(d, 3);
            x /= 10;
        }
        return sum == n;
    }
}

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        Solution sol = new Solution();
        System.out.println(sol.armstrongNumber(n));
        sc.close();
    }
}

File: Day05/Easy/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool armstrongNumber(int n) {
        int sum = 0, x = n;
        while (x > 0) {
            int d = x % 10;
            sum += pow(d, 3);
            x /= 10;
        }
        return sum == n;
    }
};

int main() {
    int n;
    cin >> n;
    Solution sol;
    cout << (sol.armstrongNumber(n) ? "true" : "false") << endl;
    return 0;
}
